api/types/resource_ids.go

```diff
@@ -81,14 +81,14 @@ func ResourceIDToString(id ResourceID) string {
 // have been obtained from ResourceIDToString.
 func ResourceIDFromString(raw string) (ResourceID, error) {
 	if len(raw) < 1 || raw[0] != '/' {
-		return ResourceID{}, trace.BadParameter("%s is not a valid ResourceID string", raw)
+		return ResourceID{}, trace.BadParameter("%s is not a valid ResourceID string. Expected /cluster/type/resource format", raw)
 	}
 	raw = raw[1:]
 	// Should be safe for any Name as long as the ClusterName and Kind don't
 	// contain slashes, which should never happen.
 	parts := strings.SplitN(raw, "/", 3)
 	if len(parts) != 3 {
-		return ResourceID{}, trace.BadParameter("/%s is not a valid ResourceID string", raw)
+		return ResourceID{}, trace.BadParameter("/%s is not a valid ResourceID string. Expected /cluster/type/resource format", raw)
```

Comment:
Maybe we can combine both checks to something like
```go
func ResourceIDFromString(raw string) (ResourceID, error) {
	// Should be safe for any Name as long as the ClusterName and Kind don't
	// contain slashes, which should never happen.
	parts := strings.SplitN(raw, "/", 4)
	if len(raw) < 1 || raw[0] != '/' || len(parts) != 4 {
		return ResourceID{}, trace.BadParameter("%q is not a valid ResourceID string. Expected /cluster/type/resource format", raw)
	}
	parts = parts[1:]
	resourceID := ResourceID{
        ...
}
```

Reply:
thanks @AntonAM , updated to this.